### PR TITLE
[Cleanup] Fix issues relating to ambiguous virtual function overloads

### PR DESCRIPTION
**Pull request by peter-b**
When a base class declares a method `virtual X::f(int)`, and a derived class declares a method `virtual Y::f(void)`, the derived class's incompatibly overloaded `f()` "masks" the base class's `f()`.  It makes it impossibly to call `Y y; Y.f(0)`, for example; an explicit cast to `X` would be required.  In these situations, recent versions of clang (such as the Emscripten compiler) provides a "overloaded-virtual" warning.

This pull request fixes all "overloaded-virtual" warnings in the Emscripten build, and turns "overloaded-virtual" warnings into errors.

### CODE REVIEW

**Comment by peter-b** [comment]
@livecode-vulcan review not ok 4609755

Failed tests locally.

**Comment by livecode-vulcan** [comment]
💔 review by @peter-b not ok 4609755d44efff5cc154e91572b7a3af13471147

**Comment by peter-b** [comment]
Pull request updated.

File: engine/src/mbldc.h [hunk 1]
```diff
@@ -195,7 +195,11 @@ class MCScreenDC: public MCUIDC
 	//
 
 	// SN-2014-07-11: [[ Bug 12769 ]] Update the signature - the non-implemented UIDC dodragdrop was called otherwise
-	MCDragAction dodragdrop(Window w, MCPasteboard *p_pasteboard, MCDragActionSet p_allowed_actions, MCImage *p_image, const MCPoint* p_image_offset);
+	/* The mobile MCScreenDC adds an overload of MCUIDC::dodragdrop()
+	 * which uses a pasteboard. By default, both do nothing
+	 * (i.e. always return DRAG_ACTION_NONE). */
+	using MCUIDC::dodragdrop;
+	virtual MCDragAction dodragdrop(Window w, MCPasteboard *p_pasteboard, MCDragActionSet p_allowed_actions, MCImage *p_image, const MCPoint* p_image_offset);
```
Comment (runrevmark) [inline engine/src/mbldc.h]: This shouldn't be necessary - more likely the mobile signature just hasn't been updated. DragDrop doesn't exist on mobile.

**Comment by peter-b** [comment]
@runrevmark Pull request updated to remove that `dodragdrop()` method entirely.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 00e38b1

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 00e38b1f8d6f91e530da187f8308d98744f5bb2f

**Comment by livecode-vulcan** [comment]
😞 test failure 00e38b1f8d6f91e530da187f8308d98744f5bb2f
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1046)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1055)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1043)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1048)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1058)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1064)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1063)

**Comment by peter-b** [comment]
@livecode-vulcan retry 00e38b1

**Comment by livecode-vulcan** [comment]
😎 test success 00e38b1f8d6f91e530da187f8308d98744f5bb2f
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1052)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1061)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1049)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1054)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1064)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1069)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1069)

**Pull request merged**